### DISCUSSION TITLE
Datumsuche

### DISCUSSION THREAD

**Discussion by Gthorsten65** [geimist/synOCR#4] [Q&A]
Hallo,
Wie wir gerade festgestellt haben, gibt es ja die datumssuche über die Python Funktion, und scheinbar noch ein fallback über das bash Skript.

Das hat dazu geführt das ein Datum bei der pythonsuche nicht gefunden wurde (weg max min year), aber dann scheinbar über die Funktion in der bash.

Ist es nicht evtl sinnvoll die Suche in der bash ganz heraus zu nehmen?
Das hätte den Vorteil, daß es nur die Suche an einer Stelle gibt...

**Comment by geimist** [comment]
Prinzipiell stimme ich dem zu. Allerdings ist die umfangreiche Suche über das Pythonmodul wesentlich zeitintensiver. Mein Testdokument benötigt dafür 17 Sekunden, während der kurze RegEx Test innerhalb von synOCR.sh im Bruchteil einer Sekunde erledigt ist. Ich weiß, dass es Einsatzszenarien von synOCR gibt, die tausende Laufzettel in kurzer Zeit zu bewältigen haben und keinen Wert auf eine möglichst genaue Datumserkennung legen.

**Mein Vorschlag:** 
Der Fallback kommt raus (zumal er inkonsistent mit den erweiterten Parametern ist). Die Auswahl in der GUI, welche Methode der User bevorzugt, würde ich aber beibehalten.

Wäre das eine Lösung?

**Reply by Gthorsten65** [reply]
Ok, du hast recht. Das oy Modul ist sehr langsam.
Wobei danke nochmal for den Hinweis. Ich wollte mit nämlich immer schon mal ansehen was da so langsam ist.
Aber dein Vorschlag kann so die Lösung sein.
Also nur eine reine regex Suche, ohne Parameter, und ansonsten das py Modul.
Der User hat die Auswahl und muss entscheiden.

**Reply by geimist** [reply]
Ich lasse in Zukunft die exemplarischen Datumsangaben (welche ungültig sind und daher für das Pythonskript für Verwirrung sorgen) herausfiltern. Damit tritt bei mir auch keine Denkpause von einigen Sekunden mehr auf.

**Comment by geimist** [comment]
closed by https://github.com/geimist/synOCR/commit/d6855f7573c415a8602cd2adc2bdd263899ad85e

**Comment by Gthorsten65** [comment]
Supi....

18.04.2023 16:17:25 Stephan Geisler ***@***.***>:
